ray-operator/controllers/ray/raycluster_controller_test.go

```diff
@@ -463,6 +498,46 @@ var _ = Context("Inside the default namespace", func() {
 			})
 			Expect(err).NotTo(HaveOccurred(), "Failed to update RayCluster")
 
+			// Make sure the suspending process is atomic by removing finalizers one by one.
+			if withConditionEnabled {
+				// Make sure the cluster is now marked with RayClusterSuspending.
+				Eventually(findRayClusterSuspendStatus, time.Second*3, time.Millisecond*500).Should(Equal(rayv1.RayClusterSuspending))
+				// Then turn the Spec.Suspend to false.
+				err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+					Eventually(
+						getResourceFunc(ctx, client.ObjectKey{Name: rayCluster.Name, Namespace: namespace}, rayCluster),
+						time.Second*3, time.Millisecond*500).Should(BeNil(), "rayCluster: %v", rayCluster)
+					suspend := false
+					rayCluster.Spec.Suspend = &suspend
+					return k8sClient.Update(ctx, rayCluster)
+				})
+				Expect(err).NotTo(HaveOccurred(), "Failed to update RayCluster")
+
+				// Remove the finalizers one by one to make sure the suspending process is still ongoing.
```

Comment:
Can we separate the atomicity test into its own test? I want to avoid mixing both code paths in a single test. In the atomicity test, can you verify the following:

1. Add finalizers to the Pods.
2. Set `suspend` to true, and check that the `suspending` condition is true.
3. Set `suspend` back to false, and ensure the `suspending` condition remains true for 2 to 3 seconds (use `Consistently`).
4. Remove the finalizers, and verify that both `suspending` and `suspended` conditions become false, all old Pods are deleted, and new Pods are created.
5. Set `suspend` to true, and all Pods should be deleted.

Reply:
ok, a new test is added.